code/modules/tgui/states/strictly_adjacent.dm

```diff
@@ -0,0 +1,20 @@
+/**
+ * tgui state: strictly_adjacent_state
+ *
+ * In addition to not incapacitated checks, only allows the interface to
+ * be open for an adjacent user. Will close the UI if adjacency check fails.
+ * There is no middleground for UI_UPDATE. This is a strict adjacency check.
+ * Ignores TK.
+ */
```

Comment:
I would suggest creating a status helper proc (for use in ui_status), instead of a new state, this is a part of our refactoring process. Mothblocks has implemented one such helper for tgui strip menu.

Reply:
I have made ze changes to utilise composition in ui_status with GLOB.always_state for ui_state